apps/arweave/src/ar_storage_module.erl

```diff
@@ -193,29 +193,139 @@ get_size(ID, [Module | Modules]) ->
 			get_size(ID, Modules)
 	end.
 
-%% @doc Return a configured storage module covering the given Offset, preferably
-%% with the given Packing. Return not_found if none is found.
+%%--------------------------------------------------------------------
+%% @doc Return a configured storage module covering the given Offset,
+%% preferably with the given Packing. Return `not_found' if none is
+%% found.
+%%
+%% == Examples ==
+%%
+%% ```
+%% ar_storage_module:get(51*3600000000001, unpacked).
+%% % {3600000000000,51,unpacked}
+%%
+%% ar_storage_module:get(11, unpacked).
+%% % not_found
+%% '''
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec get(Offset, Packing) -> Return when
+	Offset :: pos_integer(),
+	Packing :: atom(),
+	Return :: {BucketSize, Bucket, Packing}
+	        | not_found,
+	BucketSize :: pos_integer(),
+	Bucket :: pos_integer(),
+	Packing :: atom().
+
 get(Offset, Packing) ->
 	{ok, Config} = application:get_env(arweave, config),
 	get(Offset, Packing, Config#config.storage_modules, not_found).
 
-%% @doc Return the list of all configured storage modules covering the given Offset.
+%%--------------------------------------------------------------------
+%% @doc Return the list of all configured storage modules covering the
+%% given Offset. This offset is defined by the size of the storage
+%% module (in bytes) and its bucket id (e.g. 51).
+%%
+%% @see get_all/2
+%%
+%% == Examples ==
+%%
+%% We assume the application is configured with storage module `51'
+%% and `52'. This value can be found in `#config{}' record
+%%
+%% ```
+%% StorageModuleId = 52.
+%% OffsetInBytes = 3600000000000.
+%% Offset = StorageModuleId * OffsetInBytes.
+%% ar_storage_module:get_all(Offset).
+%% % [{3600000000000,51,unpacked}]
+%% '''
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec get_all(Offset) -> Return when
+	Offset :: pos_integer(),
+	Return :: [{BucketSize, Bucket, Packing}],
+	BucketSize :: pos_integer(),
+	Bucket :: pos_integer(),
+	Packing :: atom().
+
 get_all(Offset) ->
 	{ok, Config} = application:get_env(arweave, config),
 	get_all(Offset, Config#config.storage_modules, []).
 
-%% @doc Return the list of configured storage modules whose ranges intersect
-%% the given interval.
+%%--------------------------------------------------------------------
+%% @doc Return the list of configured storage modules whose ranges
+%% intersect the given interval.
+%%
+%% @see get_all/1
+%%
+%% == Examples ==
+%%
+%% ```
+%% StartOffset = 51*3600000000001.
```

Comment:
ditto comment above. The offsets aren't generally defined in terms of partition boundaries or storage module boundaries. They are generally offsets of bytes or chunks in the weave.